Controls/Mail/Mailman.c

```diff
@@ -13,6 +16,26 @@
  */
 void depositLetter(void) {
 	OS_ERR err; // Make sure to check for errors and print the error code if not OS_ERR_NONE
+
+	// Generate Random Number
+	char randomNumberString[4];
```

Comment:
The biggest number you generate is 9999. How many bytes of space will it take up as a string?

Reply:
A char stores 1 byte of information, so since this string is 4 characters long, the number string will take up 4 bytes of memory.

Reply:
So if a string is just a sequence of characters, how do you know where the end of the string is? How would the program know?

Reply:
The end of the string should have a null termination character (\0). This means there will actually be an element in the 4th index of \0, meaning the total storage would be 5 bytes of memory.
The way the code was originally written was not optimal, though. sprintf wrote a string of 4 characters into a char array with a length of 4. I think this means the null terminator character was not appended. This could have led to some undefined behavior? I was noticing a strange bug that disappeared after I changed the way the message was posted to the mailbox.

Reply:
The nul terminator was still appended, but it would've gone into whatever memory is right after the array. That could cause weird behavior depending on what's there.

Reply:
That makes sense. Thanks for the clarification.